### ISSUE TITLE
Для всех: Отладка кода

### ISSUE THREAD

**Issue by taqyon** [atls/convention#25]

**Comment by taqyon** [comment]
@TFK70 @TorinAsakura 

[Добавил файл отладки кода.](https://github.com/atls/convention/blob/master/docs/Debugging.md) 

Я кроме форматирования и стилистики ничего не менял. Суть осталась та же. Посмотрите и проверьте всё ли на месте.

**Comment by taqyon** [comment]
Я также добавил секцию советы: с целью поместить туда внешние или внутренние ссылки. 

Например у меня есть такая инфа: 

# Почему нельзя загружать изображения кода / ошибок, когда задаешь вопрос?

Ты не должен размещать в виде изображения: 
- код 
- сообщения об ошибках
- исключениях
- файлы журналов
- файлы конфигурации
- файлы проекта 
- или что-либо еще, представленное в текстовой форме 

## Потому что
- Код или примеры данных на изображениях нельзя скопировать и вставить в редактор и скомпилировать, чтобы воспроизвести проблему
- Изображения имеют большой размер и трудночитаемы на мобильных устройствах
- Изображения не могут быть использованы для поиска
- Изображения труднее читать, чем текст
- Постить изображения твоего кода сложнее, чем копировать/вставлять сам код и форматировать его
- Ты просишь помощи с проблемой и должен максимально облегчить задачу помощи
- URL-адреса размещенных изображений часто являются недоступными
- Изображения показывают ограниченное количество строк кода. Для умеренно сложных вопросов невозможно уместить весь необходимый код на одном экране, даже если ты создал самый минимально возможный пример для воспроизведения рассматриваемого вопроса
- Помощник не сможет понять, вызвана ли ошибка в вашем коде, скажем, невидимыми символами или неправильным использованием символов Unicode, которые выглядят одинаково
- Некоторые люди используют темную тему для просмотра сайта. Белый фон изображения может повредить их глазам при просмотре сайта в позднее время. Кроме того, изображение с прозрачным фоном может быть отлично видно в светлой теме, но едва различимо в темной

 Нужно еще? :-)

**Изображения следует использовать только для иллюстрации проблем, которые не могут быть понятны другим способом, например, для предоставления скриншотов пользовательского интерфейса.**


---

Это переработанная инфа с Stackoverflow

**Comment by TFK70** [comment]
@taqyon все на месте, но на мой взгляд здесь нужно изменить несколько вещей:

### Насчет шаблонов

Сюда точно нужно добавить md шаблон который люди должны использовать при обращении за помощью. Если этого не делать - первым вопросом человека будет что-то типа: `Ну вот у меня кароч в какой-то консоли вот это вывело <какой-то непонятный лог> хз че это я потыкал чет не понял помогите` **даже** если сообщение было опубликовано на гитхабе, в телеге о помощи просят еще хуже

В конечном итоге обращения должны выглядеть примерно [так](https://github.com/atls-academy/brusselboy.github.io/issues/12#issuecomment-1158823075). По идее за пример можно взять любое сообщение из треда или одной из других последних задач в репе Кирилла

Я уже пытался подготовить на этой почве шаблон обращения за помощью, получилось что-то [такое](https://gist.github.com/TFK70/07f469bb31e64481b22d0f66d43c8b88) (во многом дублирует [этот](https://github.com/atls-academy/brusselboy.github.io/blob/master/.github/ISSUE_TEMPLATE/bug.md) шаблон)

В целом можно использовать и его, если он на твой и @TorinAsakura взгляд валиден, **но** главное упомянуть, что ни один из пунктов не должен быть проигнорирован, человек должен хотя бы прокомментировать отсутствие того или иного пункта

### Насчет секции "Локализация проблемы"

Тут скорее к @TorinAsakura 

Предлагаю эту секцию либо убрать, либо переписать, чтобы протокол подходил под гораздо большее кол-во кейсов

На данный момент протокол работает только на кейсах, когда ошибка непосредственно в проектном файле. Все что идет извне протокол уже не покрывает, скорее наоборот вводит в заблуждение

### Добавить пункт "Обращение за помощью"

Сюда я и предлагаю поместить [этот](https://github.com/atls-academy/brusselboy.github.io/issues/12#issuecomment-1158823075) шаблон и описать это:

**При обращении за помощью нужно:**

1. Сделать коммит с текущим состоянием кода, где воспроизводится проблема
2. Открыть draft pull request, если еще не открыт
3. Обратиться за помощью заполнив [этот](https://github.com/atls-academy/brusselboy.github.io/issues/12#issuecomment-1158823075) шаблон

Форматирование и тп разумеется не принципиальны абсолютно, главное эту инфу донести

Это чисто мое видение, так что критикуйте в случае чего

**Comment by taqyon** [comment]
> Сюда точно нужно добавить md шаблон который люди должны использовать при обращении за помощью.

Речь о шаблоне issue?

**Comment by taqyon** [comment]
> ### Насчет шаблонов
> 
> Сюда точно нужно добавить md шаблон который люди должны использовать при обращении за помощью. Если этого не делать - первым вопросом человека будет что-то типа: `Ну вот у меня кароч в какой-то консоли вот это вывело <какой-то непонятный лог> хз че это я потыкал чет не понял помогите` **даже** если сообщение было опубликовано на гитхабе, в телеге о помощи просят еще хуже
> 
> В конечном итоге обращения должны выглядеть примерно [так](https://github.com/atls-academy/brusselboy.github.io/issues/12#issuecomment-1158823075). По идее за пример можно взять любое сообщение из треда или одной из других последних задач в репе Кирилла
> 
> Я уже пытался подготовить на этой почве шаблон обращения за помощью, получилось что-то [такое](https://gist.github.com/TFK70/07f469bb31e64481b22d0f66d43c8b88) (во многом дублирует [этот](https://github.com/atls-academy/brusselboy.github.io/blob/master/.github/ISSUE_TEMPLATE/bug.md?rgh-link-date=2022-06-20T16%3A21%3A04Z) шаблон)
> 
> В целом можно использовать и его, если он на твой и @TorinAsakura взгляд валиден, **но** главное упомянуть, что ни один из пунктов не должен быть проигнорирован, человек должен хотя бы прокомментировать отсутствие того или иного пункта

Я перечитал твой ответ несколько раз. Может стоит это вынести в отдельный issue? И там продолжить обсуждение?

**Comment by TFK70** [comment]
@taqyon да, как удобнее

Написал сюда потому что на мой взгляд эта инфа именно в этом разделе должна быть

**Comment by TorinAsakura** [comment]
@taqyon Да. Это отличная тема, постоянно приходится это повторять по кругу всем новичкам.
